p2p/node/localnodestore.go

```diff
@@ -23,114 +21,77 @@ type nodeFileData struct {
 
 // PersistData save node's data to local disk in `path`.
 func (n *LocalNode) PersistData(path string) error {
-
 	data := nodeFileData{
 		PubKey:  n.publicKey.String(),
 		PrivKey: n.privKey.String(),
 	}
 
 	finaldata, err := json.MarshalIndent(data, "", "  ")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to marshal data: %w", err)
 	}
 
 	datadir := filepath.Join(path, config.P2PDirectoryPath, config.NodesDirectoryName, n.publicKey.String())
-
-	err = filesystem.ExistOrCreate(datadir)
-	if err != nil {
-		return err
+	if err = filesystem.ExistOrCreate(datadir); err != nil {
+		return fmt.Errorf("failed to check or create datadir: %w", err)
 	}
 
 	nodefile := filepath.Join(datadir, config.NodeDataFileName)
 
 	// make sure our node file is written to the os filesystem.
-	f, err := os.Create(nodefile)
-	if err != nil {
-		return err
-	}
-
-	_, err = f.Write(finaldata)
-	if err != nil {
-		return err
-	}
-
-	err = f.Sync()
-	if err != nil {
-		return err
-	}
-
-	err = f.Close()
-	if err != nil {
-		return err
+	if err := ioutil.WriteFile(nodefile, finaldata, 0666); err != nil {
+		return fmt.Errorf("failed to write file: %w", err)
 	}
 
+	// TODO(josebalius): should we accept a logger or otherwise not log directly
```

Comment:
This log call feels like it's best done by callers or at the very least we should accept a `logger`